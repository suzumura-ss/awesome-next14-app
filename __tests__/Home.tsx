import React from 'react';
import { render } from '@testing-library/react';
import Home from '@src/app/page';

jest.mock('@backend/todo');

async function resolvedComponent<T>(Component: (props: T) => Promise<React.JSX.Element>, props: T) {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
}

describe(Home, () => {
  it('renders correctly', async () => {
    const HomeResolved = await resolvedComponent(Home, {});

    const { container } = render(<HomeResolved />);
    expect(container).toMatchSnapshot();
  });
});
