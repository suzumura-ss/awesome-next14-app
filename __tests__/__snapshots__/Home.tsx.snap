// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <header
    class="flex justify-between items-center mb-4"
  >
    <h1
      class="text-2xl"
    >
      ToDos
    </h1>
    <a
      class="border border-slate-300 text-slate-300 px-2 py-1 rounded hover:bg-slate-700 focus-within:bg-slate-700 outline-none"
      href="/new"
    >
      New
    </a>
  </header>
  <ul
    class="pl-4"
  />
</div>
`;
